### PR TITLE
Avoid deprecated `progress_bar_refresh_rate` usage

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Part of #9939

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @awaelchli

### CODE REVIEW

File: tests/callbacks/test_tqdm_progress_bar.py [hunk 1]
```diff
@@ -32,54 +32,40 @@
...
-        ([], None),
-        ([], 1),
-        ([], 2),
-        ([TQDMProgressBar(refresh_rate=1)], 0),
-        ([TQDMProgressBar(refresh_rate=2)], 0),
-        ([TQDMProgressBar(refresh_rate=2)], 1),
+        # won't print but is still set
+        {"callbacks": TQDMProgressBar(refresh_rate=0)},
+        {"callbacks": TQDMProgressBar()},
+        # callbacks takes precedence
```
Comment (carmocca) [inline tests/callbacks/test_tqdm_progress_bar.py:40]: I'm assuming that passing the instance as a callback should take precedence over `enable_progress_bar`

File: tests/callbacks/test_tqdm_progress_bar.py [hunk 2]
```diff
@@ -276,20 +263,21 @@ def test_tqdm_progress_bar_default_value(tmpdir):
     trainer = Trainer(default_root_dir=tmpdir)
     assert trainer.progress_bar_callback.refresh_rate == 1
 
-    trainer = Trainer(default_root_dir=tmpdir, progress_bar_refresh_rate=None)
```
Comment (carmocca) [inline tests/callbacks/test_tqdm_progress_bar.py:279]: This is exactly the same as in the assertion above

File: tests/callbacks/test_tqdm_progress_bar.py [hunk 3]
```diff
@@ -276,20 +263,21 @@ def test_tqdm_progress_bar_default_value(tmpdir):
...
 @mock.patch.dict(os.environ, {"COLAB_GPU": "1"})
 def test_tqdm_progress_bar_value_on_colab(tmpdir):
     """Test that Trainer will override the default in Google COLAB."""
     trainer = Trainer(default_root_dir=tmpdir)
     assert trainer.progress_bar_callback.refresh_rate == 20
 
-    trainer = Trainer(default_root_dir=tmpdir, progress_bar_refresh_rate=None)
-    assert trainer.progress_bar_callback.refresh_rate == 20
+    trainer = Trainer(default_root_dir=tmpdir, callbacks=TQDMProgressBar())
+    assert trainer.progress_bar_callback.refresh_rate == 1  # FIXME: should be 20
```
Comment (carmocca) [inline tests/callbacks/test_tqdm_progress_bar.py:274]: Is this a bug? Can be done in a separate PR.

Comment (tchaton) [reply tests/callbacks/test_tqdm_progress_bar.py:274]: Yes and no :) As you provide manual the progress bar, the default value is set to 1. However, it would be better to set it to 20 in the COLAB env.

I believe a way to resolve this would be for this logic to live within the Callback directly as follows:

```py
import os
from typing import Optional

class TQDMProgressBar(ProgressBarBase):

    def __init__(self, refresh_rate: Optional[int] = None, process_position: int = 0):
        super().__init__()

        if refresh_rate is None:
            refresh_rate = 20 if os.getenv("COLAB_GPU", "1") == "1" else 1

        self._refresh_rate = refresh_rate
```

Comment (carmocca) [reply tests/callbacks/test_tqdm_progress_bar.py:274]: Yes, exactly.

File: pytorch_lightning/trainer/connectors/callback_connector.py [hunk 4]
```diff
@@ -94,12 +94,9 @@ def on_trainer_init(
                 " bar pass `enable_progress_bar = False` to the Trainer."
             )
 
-        if enable_progress_bar:
-            self.trainer._progress_bar_callback = self.configure_progress_bar(
-                progress_bar_refresh_rate, process_position
-            )
-        else:
-            self.trainer._progress_bar_callback = None
+        self.trainer._progress_bar_callback = self.configure_progress_bar(
```
[the remark this answers is not in the document]
Comment (carmocca) [reply pytorch_lightning/trainer/connectors/callback_connector.py:97]: This is a fix so i'd rather not change the API here.

File: tests/callbacks/test_tqdm_progress_bar.py (hunk 3 above)
Comment (kaushikb11) [reply tests/callbacks/test_tqdm_progress_bar.py:274]: Added it here #10542. And tested that `RichProgressBar` wouldn't require this setup

**Review by kaushikb11 — APPROVED** [review]

File: tests/callbacks/test_tqdm_progress_bar.py (hunk 3 above)
Comment (carmocca) [reply tests/callbacks/test_tqdm_progress_bar.py:276]: Thanks @kaushikb11, let's get this merged so you can fix it on top

**Review by tchaton — APPROVED** [review]
LGTM !

**Comment by daniellepintz** [comment]
Hmm, are we sure this is the desired behavior here? If a user explicitly overrides the default and sets `enable_progress_bar=False` it is weird that we still enable a progress bar.

File: pytorch_lightning/trainer/connectors/callback_connector.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ananthsub) [reply pytorch_lightning/trainer/connectors/callback_connector.py:97]: 1. The model checkpoint callback raises a misconfiguration error if the flags are inconsistent (`enable_checkpointing=False` yet `callbacks` contains at least one `ModelCheckpoint` callback instance). IMO we should have the same behavior for progress bars for consistency.
2. (for a separate PR) I'm not sure why `progress_bar_callback` has a separate property on the Trainer. I'd prefer defining a property like this which is derived from the `callbacks` property: https://github.com/PyTorchLightning/pytorch-lightning/blob/40945a51765e72400dd3be8f1a84a52a69effa73/pytorch_lightning/trainer/trainer.py#L1841-L1846

**Comment by carmocca** [comment]
> Hmm, are we sure this is the desired behavior here? If a user explicitly overrides the default and sets enable_progress_bar=False it is weird that we still enable a progress bar.

The user sends a mixed signal, both passing it as a callback and disabling it, we can go for either of them.

**Comment by daniellepintz** [comment]
I agree with @ananthsub, I think we should just raise a MisConfigException

File: pytorch_lightning/trainer/connectors/callback_connector.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (carmocca) [reply pytorch_lightning/trainer/connectors/callback_connector.py:97]: I'll update the behaviour on conflict.

[the remark this answers is not in the document]
Comment (carmocca) [reply pytorch_lightning/trainer/connectors/callback_connector.py:97]: @ananthsub feel free to create an issue for (2)

File: CHANGELOG.md [hunk 5]
```diff
@@ -134,7 +134,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Squeeze the early stopping monitor to remove empty tensor dimensions ([#10461](https://github.com/PyTorchLightning/pytorch-lightning/issues/10461))
 
 
--
+- Fixed issue where `trainer.progress_bar_callback` was not getting set with `enable_progress_bar=False` if a progress bar instance has been passed in the callback list ([#10520](https://github.com/PyTorchLightning/pytorch-lightning/issues/10520))
```
Comment (daniellepintz) [inline CHANGELOG.md:137]: update this to added misconfig check when `enable_progress_bar=False` and pbar instance has been passed?

**Review by daniellepintz — APPROVED** [review]
Looks good!

File: CHANGELOG.md [hunk 6]
```diff
@@ -31,7 +31,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - The `monitor` argument in the `EarlyStopping` callback is no longer optional ([#10328](https://github.com/PyTorchLightning/pytorch-lightning/pull/10328))
 
 
--
+- Raise `MisconfigurationException` with `enable_progress_bar=False` if a progress bar instance has been passed in the callback list ([#10520](https://github.com/PyTorchLightning/pytorch-lightning/issues/10520))
```
Comment (daniellepintz) [inline CHANGELOG.md:34]:
```suggestion
- Raise `MisconfigurationException` when `enable_progress_bar=False` and a progress bar instance has been passed in the callback list ([#10520](https://github.com/PyTorchLightning/pytorch-lightning/issues/10520))
```
